File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/ObjC/CRUD/CRUD/ViewControllers/SortViewController.m

#import "SortViewController.h"
#import "CodeSampleViewController.h"
#import "Utils.h"

#define FIND_SORT @"Find with sort"
#define SHOW_SAMPLE @"ShowCodeSampleFromSortBy"

@interface SortViewController() {
    NSMutableArray<NSString *> *sortBy;
}
@end

@implementation SortViewController

- (void)viewDidLoad {
    [super viewDidLoad];    
    sortBy = [NSMutableArray<NSString *> new];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.fields.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"SortCell" forIndexPath:indexPath];
    cell.textLabel.text = [self.fields objectAtIndex:indexPath.row];
    return cell;
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    CodeSampleViewController *codeSampleVC = [segue destinationViewController];
    codeSampleVC.tableName = self.tableName;
    codeSampleVC.operation = FIND_SORT;
    codeSampleVC.sortBy = sortBy;
}

- (IBAction)next:(id)sender {
    if (sortBy.count > 0) {
        [self performSegueWithIdentifier:SHOW_SAMPLE sender:nil];
    }
    else {
        Fault *fault = [[Fault alloc] initWithMessage:@"Please select fields to sort by" faultCode:0];
        [utils showErrorAlert:self fault:fault];
    }
}

- (IBAction)selectProperty:(id)sender {
    if ([[[sender superview] superview] isKindOfClass:[UITableViewCell class]]) {
        UITableViewCell *cell = (UITableViewCell *)[[sender superview] superview];
        NSIndexPath *indexPath = [self.tableView indexPathForCell:cell];
        NSString *property = [self.fields objectAtIndex:indexPath.row];
        if ([(UISwitch *)sender isOn]) {
            [sortBy addObject:property];
        }
        else {
            [sortBy removeObject:property];
        }
    }
}

@end

File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/Swift/CRUD/CRUD/ViewControllers/SortViewController.swift

import UIKit
import Backendless

class SortViewController: UITableViewController {
    
    var tableName = ""
    var fields = [String]()
    
    private var sortBy = [String]()
    
    private let FIND_SORT = "Find with sort"
    private let SHOW_SAMPLE = "ShowCodeSampleFromSortBy"
    private let utils = Utils.shared

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return fields.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "SortCell", for: indexPath)
        cell.textLabel?.text = fields[indexPath.row]
        return cell
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if let codeSampleVC = segue.destination as? CodeSampleViewController {
            codeSampleVC.tableName = tableName
            codeSampleVC.operation = FIND_SORT
            codeSampleVC.sortBy = sortBy
        }
    }
    
    @IBAction func next(_ sender: Any) {
        if sortBy.count > 0 {
            performSegue(withIdentifier: SHOW_SAMPLE, sender: nil)
        }
        else {
            let fault = Fault(message: "Please select fields to sort by", faultCode: 0)
            utils.showErrorAler(target: self, fault: fault)
        }
    }
    
    @IBAction func selectProperty(_ sender: Any) {
        if let sender = sender as? UISwitch,
            let cell = sender.superview?.superview as? UITableViewCell,
            let indexPath = tableView.indexPath(for: cell) {
            let property = fields[indexPath.row]
            if sender.isOn {
                sortBy.append(property)
            }
            else if let index = sortBy.firstIndex(of: property) {
                sortBy.remove(at: index)
            }
        }
    }
}
